google/cloud/bigtable/client_options.h

```diff
@@ -177,13 +177,13 @@ class ClientOptions {
 
   /// Access all the channel arguments.
   grpc::ChannelArguments channel_arguments() const {
-    return channel_arguments_;
+    return ::google::cloud::internal::MakeChannelArguments(opts_);
   }
 
   /// Set all the channel arguments.
   ClientOptions& set_channel_arguments(
       grpc::ChannelArguments const& channel_arguments) {
-    channel_arguments_ = channel_arguments;
+    opts_.set<GrpcChannelArgumentsNativeOption>(channel_arguments);
```

Comment:
I considered unsetting `GrpcChannelArgumentsOptions` here.
```
opts_.unset<GrpcChannelArgumentsOption>();
```

The `grpc::ChannelArguments` object that gets made includes both options (and `UserAgentProductsOption`). So we could have a situation like this:
```
ClientOptions client_options(Options{}.set<GrpcChannelArgumentsOption>({{"key", "value"}});
client_options.set_channel_arguments(channel_args);
auto returned = client_options.channel_arguments();
// returned != channel_args
// because `returned` contains the extra ("key", "value") pair.
```

But I know we don't want to unset `UserAgentProductsOption`. And it's a bit weird to unset `GrpcChannelArgumentsOption` and not `UserAgentProductsOption`.

I suspect this is an inconsequential difference, as we will recommend against these methods..... but if someone else thinks differently, I would be happy to unset the map option.

Reply:
Ack